request.js

```diff
@@ -1943,7 +1961,11 @@ Request.prototype.end = function (chunk) {
   if (self.req) {
     self.req.end()
 
-    self.req._header && (self._reqResInfo.request.headers = parseRequestHeaders(self.req._header))
+    // Reference to request, so if _reqResInfo is updated (in case of redirects), we still can update the headers
+    const request = self._reqResInfo.request
+    Promise.resolve(self.req._header).then(function (header) {
+      request.headers = parseRequestHeaders(header)
```

Comment:
Not needed, since once the `_header` value is resolved, it means that we have access to the final request object that was sent, thus subsequent gets for `_header` will return the value instead of the promise. Additionally, since `_header` is implemented as a getter, it cannot be reassigned